src/main/java/io/opentracing/contrib/tracerresolver/TracerConverter.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.opentracing.contrib.tracerresolver;
+
+import io.opentracing.Tracer;
+
+/**
+ * Function converting an existing {@link Tracer}.
+ * <p>
+ * This can be useful for <em>wrapping</em> tracers:
+ * <pre><code>
+ * public final class FooWrapperConverter implements TracerConverter {
+ *     public Tracer convert(Tracer existingTracer) {
+ *         return new FooTracerWrapper(existingTracer);
+ *     }
+ * }
+ * </code></pre>
+ *
+ * @author Sjoerd Talsma
+ */
+public interface TracerConverter {
+
+    /**
+     * Function that converts a {@link Tracer}.
+     * <p>
+     * It may either manipulate the tracer or return an entirely new {@linkplain Tracer} instance.
+     *
+     * @param existingTracer The existing tracer to be converted
+     *                       (might be <code>null</code>, e.g. if no tracer could be resolved).
```

Comment:
Of course it could be handled that way. I guess I'm lazy :wink: 
Will dabble a little as I can't (now) come up with a usecase for a `null` -> non-`null` conversion.